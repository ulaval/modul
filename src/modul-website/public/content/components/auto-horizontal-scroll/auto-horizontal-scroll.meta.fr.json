{
    "name": "DÃ©filement horizontal automatique",
    "url": "defilement-horizontal-automatique",
    "category": "layout",
    "type": "components",
    "beta": true,
    "visible": true,
    "preview": "components/auto-horizontal-scroll/auto-horizontal-scroll.preview.fr.md",
    "components": [
        "m-auto-horizontal-scroll"
    ]
}
