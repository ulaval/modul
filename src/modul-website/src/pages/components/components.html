<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
