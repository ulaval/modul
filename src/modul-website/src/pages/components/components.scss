@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
