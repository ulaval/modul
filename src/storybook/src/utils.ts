export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
