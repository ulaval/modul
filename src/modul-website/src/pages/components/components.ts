
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
