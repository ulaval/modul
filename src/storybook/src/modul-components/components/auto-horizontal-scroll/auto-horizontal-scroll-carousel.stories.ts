import { boolean, select } from '@storybook/addon-knobs';
import { MAutoHorizontalScrollGradientStyle } from '@ulaval/modul-components/dist/components/auto-horizontal-scroll/auto-horizontal-scroll';
import { AUTO_HORIZONTAL_SCROLL } from '@ulaval/modul-components/dist/components/component-names';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${AUTO_HORIZONTAL_SCROLL}/CarouselExample`,
    parameters: { fileName: __filename }
};

const getCarouselTemplate = (
    _horizontalScrollOffset: number,
    _nbItem: number,
    _displayButton: boolean
) => ({
    props: {
        leftGradientActive: {
            default: boolean('Prop left-gradient-active', true)
        },
        rightGradientActive: {
            default: boolean('Prop right-gradient-active', true)
        },
        previousButtonActive: {
            default: boolean('Prop previous-button-active', _displayButton)
        },
        nextButtonActive: {
            default: boolean('Prop next-button-active', _displayButton)
        },
        gradientStyle: {
            default: select(
                'Prop gradient-style',
                Enums.toValueArray(
                    MAutoHorizontalScrollGradientStyle
                ),
                MAutoHorizontalScrollGradientStyle.White
            )
        },
        displayHorizontalScrollbar: {
            default: boolean('Prop display-horizontal-scrollbar', true)
        }
    },
    data: () => ({
        horizontalScrollOffset: _horizontalScrollOffset,
        crouselItemSize: 200,
        crouselItemSpacing: 16,
        nbItem: _nbItem
    }),
    methods: {
        isActiveBox(horizontalScrollOffset: number, boxIndex: number): boolean {
            const _this: any = this;
            const boxFullWidth: number = _this.crouselItemSize + _this.crouselItemSpacing;
            return horizontalScrollOffset >= (((boxIndex - 1) * boxFullWidth) - 160)
                && horizontalScrollOffset < (((boxIndex - 1) * boxFullWidth) + boxFullWidth - 160);
        },
        onPreviousButtonClick(): void {
            const _this: any = this;
            _this.incrementHorizontalScrollOffset(-_this.crouselItemSize - _this.crouselItemSpacing);
        },
        onNextButtonClick(): void {
            const _this: any = this;
            _this.incrementHorizontalScrollOffset(_this.crouselItemSize + _this.crouselItemSpacing);
        },
        incrementHorizontalScrollOffset(increment: number): void {
            const _this: any = this;
            _this.horizontalScrollOffset = _this.horizontalScrollOffset + increment;
        }
    },
    template: `<${AUTO_HORIZONTAL_SCROLL}
        :min-width="crouselItemSize * nbItem + (nbItem + 1) * crouselItemSpacing + 'px'"
        :left-gradient-active="leftGradientActive"
        :right-gradient-active="rightGradientActive"
        :previous-button-active="previousButtonActive"
        :next-button-active="nextButtonActive"
        :horizontal-scroll-offset.sync="horizontalScrollOffset"
        :gradient-style="gradientStyle"
        :display-horizontal-scrollbar="displayHorizontalScrollbar"
        style="background: #f4f4f4;"
        @previous-button-click="onPreviousButtonClick()"
        @next-button-click="onNextButtonClick()"
    >
        <div style="display: flex; padding: 32px 16px;">
            <article
                v-for="index in nbItem"
                :key="index"
                :data-box-index="index"
                style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px;"
                :style="{
                    marginRight: index < nbItem ? crouselItemSpacing + 'px' : undefined,
                    backgroundColor: isActiveBox(horizontalScrollOffset, index) ? '#ffc103': '#fff',
                    width: crouselItemSize + 'px',
                    height: crouselItemSize + 'px',
                }"
            >
                <p class="m-u--no-margin">Carousel item #{{ index }}</p>
            </article>
        </div>
    </${AUTO_HORIZONTAL_SCROLL}>`
});

export const Sandbox = () => getCarouselTemplate(1200, 20, false);

export const TwoItems = () => getCarouselTemplate(0, 2, false);
export const WithButton = () => getCarouselTemplate(120, 14, true);
