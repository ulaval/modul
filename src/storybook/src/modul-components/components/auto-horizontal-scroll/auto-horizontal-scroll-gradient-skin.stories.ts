import { MAutoHorizontalScrollGradientStyle } from '@ulaval/modul-components/dist/components/auto-horizontal-scroll/auto-horizontal-scroll';
import { AUTO_HORIZONTAL_SCROLL } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${AUTO_HORIZONTAL_SCROLL}/GradientStyle`,
    parameters: { fileName: __filename }
};

const getGradientSkinTemplate = (gradientStyle: MAutoHorizontalScrollGradientStyle) => ({
    template: `<${AUTO_HORIZONTAL_SCROLL}
        min-width="2200px"
        :previous-button-active="true"
        :next-button-active="true"
        gradient-style="${ gradientStyle}"
    >
        <div style="background: #1f1f1f; padding: 32px; color: #fff; width: 100%;">
            <p class="m-u--no-margin">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        </div>
    </${AUTO_HORIZONTAL_SCROLL}>`
});

export const WhiteBackground = () => getGradientSkinTemplate(MAutoHorizontalScrollGradientStyle.White);
export const Dark = () => getGradientSkinTemplate(MAutoHorizontalScrollGradientStyle.Dark);
export const Light = () => getGradientSkinTemplate(MAutoHorizontalScrollGradientStyle.Light);
export const Interactive = () => getGradientSkinTemplate(MAutoHorizontalScrollGradientStyle.Interactive);
