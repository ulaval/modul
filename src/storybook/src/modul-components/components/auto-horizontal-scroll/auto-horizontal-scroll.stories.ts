import { actions } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import { MAutoHorizontalScrollGradientStyle } from '@ulaval/modul-components/dist/components/auto-horizontal-scroll/auto-horizontal-scroll';
import { AUTO_HORIZONTAL_SCROLL } from '@ulaval/modul-components/dist/components/component-names';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${AUTO_HORIZONTAL_SCROLL}`,
    parameters: { fileName: __filename }
};

const getTemplate = (
    _previousButtonActive: boolean,
    _nextButtonActive: boolean,
    _leftGradientActive: boolean,
    _rightGradientActive: boolean,
    _previousButtonText: string = 'Previous',
    _nextButtonText: string = 'Next',
    _dragActive: boolean = true,
    _displayHorizontalScrollbar: boolean = true
) => ({
    props: {
        minWidth: {
            default: text('Prop min-width', '1900px')
        },
        dragActive: {
            default: boolean('Prop drag-active', _dragActive)
        },
        previousButtonActive: {
            default: boolean('Prop previous-button-active', _previousButtonActive)
        },
        nextButtonActive: {
            default: boolean('Prop next-button-active', _nextButtonActive)
        },
        previousButtonText: {
            default: text('Prop previous-button-text', _previousButtonText)
        },
        nextButtonText: {
            default: text('Prop next-button-text', _nextButtonText)
        },
        leftGradientActive: {
            default: boolean('Prop left-gradient-active', _leftGradientActive)
        },
        rightGradientActive: {
            default: boolean('Prop right-gradient-active', _rightGradientActive)
        },
        gradientStyle: {
            default: select(
                'Prop gradient-style',
                Enums.toValueArray(
                    MAutoHorizontalScrollGradientStyle
                ),
                MAutoHorizontalScrollGradientStyle.White
            )
        },
        displayHorizontalScrollbar: {
            default: boolean('Prop display-horizontal-scrollbar', _displayHorizontalScrollbar)
        },
        slotDefault: {
            default: boolean('Slot default', true)
        }
    },
    methods: actions('emitPreviousButtonClick', 'emitNextButtonClick'),
    data: () => ({
        horizontalScrollOffset: 60
    }),
    template: `<${AUTO_HORIZONTAL_SCROLL}
        :min-width="minWidth"
        :drag-active="dragActive"
        :left-gradient-active="leftGradientActive"
        :right-gradient-active="rightGradientActive"
        :previous-button-active="previousButtonActive"
        :next-button-active="nextButtonActive"
        :previous-button-text="previousButtonText"
        :next-button-text="nextButtonText"
        :gradient-style="gradientStyle"
        :horizontal-scroll-offset.sync="horizontalScrollOffset"
        :display-horizontal-scrollbar="displayHorizontalScrollbar"
        @previous-button-click="emitPreviousButtonClick"
        @next-button-click="emitNextButtonClick"
    >
        <div v-if="slotDefault"
            style="background: #1f1f1f; padding: 32px; color: #fff; width: 100%;"
        >
            <p class="m-u--no-margin">Adjust the width of your browser to <strong>{{ parseInt(minWidth, 10) - 80 }}px</strong> to be able to test the component.</p>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        </div>
    </${AUTO_HORIZONTAL_SCROLL}>`
});

export const Sandbox = () => getTemplate(true, true, true, true);

export const PreviousButtonActive = () => getTemplate(true, false, false, false);
export const NextButtonActive = () => getTemplate(false, true, false, false);

export const LeftGradientActive = () => getTemplate(false, false, true, false);
export const RightGradientActive = () => getTemplate(false, false, false, true);

export const PreviousButtonText = () => getTemplate(true, false, false, false, 'Custom Previous');
export const NextButtonText = () => getTemplate(false, true, false, false, 'Previous', 'Custom Next');

export const DragDisabled = () => getTemplate(true, true, false, false, 'Previous', 'Next', false);
export const HideHorizontalScrollbar = () => getTemplate(true, true, false, false, 'Previous', 'Next', true, false);


