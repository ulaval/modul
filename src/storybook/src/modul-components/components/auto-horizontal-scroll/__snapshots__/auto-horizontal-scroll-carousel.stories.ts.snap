// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-auto-horizontal-scroll/CarouselExample Sandbox 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-style-white"
  style="background: rgb(244, 244, 244); position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 4336px;"
    >
      <div
        style="display: flex; padding: 32px 16px;"
      >
        <article
          data-box-index="1"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #1
          </p>
        </article>
        <article
          data-box-index="2"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #2
          </p>
        </article>
        <article
          data-box-index="3"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #3
          </p>
        </article>
        <article
          data-box-index="4"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #4
          </p>
        </article>
        <article
          data-box-index="5"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #5
          </p>
        </article>
        <article
          data-box-index="6"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #6
          </p>
        </article>
        <article
          data-box-index="7"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 193, 3); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #7
          </p>
        </article>
        <article
          data-box-index="8"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #8
          </p>
        </article>
        <article
          data-box-index="9"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #9
          </p>
        </article>
        <article
          data-box-index="10"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #10
          </p>
        </article>
        <article
          data-box-index="11"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #11
          </p>
        </article>
        <article
          data-box-index="12"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #12
          </p>
        </article>
        <article
          data-box-index="13"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #13
          </p>
        </article>
        <article
          data-box-index="14"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #14
          </p>
        </article>
        <article
          data-box-index="15"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #15
          </p>
        </article>
        <article
          data-box-index="16"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #16
          </p>
        </article>
        <article
          data-box-index="17"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #17
          </p>
        </article>
        <article
          data-box-index="18"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #18
          </p>
        </article>
        <article
          data-box-index="19"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #19
          </p>
        </article>
        <article
          data-box-index="20"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #20
          </p>
        </article>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll/CarouselExample Two Items 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-style-white"
  style="background: rgb(244, 244, 244); position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 448px;"
    >
      <div
        style="display: flex; padding: 32px 16px;"
      >
        <article
          data-box-index="1"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 193, 3); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #1
          </p>
        </article>
        <article
          data-box-index="2"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #2
          </p>
        </article>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll/CarouselExample With Button 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-style-white"
  style="background: rgb(244, 244, 244); position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 3040px;"
    >
      <div
        style="display: flex; padding: 32px 16px;"
      >
        <article
          data-box-index="1"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #1
          </p>
        </article>
        <article
          data-box-index="2"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 193, 3); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #2
          </p>
        </article>
        <article
          data-box-index="3"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #3
          </p>
        </article>
        <article
          data-box-index="4"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #4
          </p>
        </article>
        <article
          data-box-index="5"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #5
          </p>
        </article>
        <article
          data-box-index="6"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #6
          </p>
        </article>
        <article
          data-box-index="7"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #7
          </p>
        </article>
        <article
          data-box-index="8"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #8
          </p>
        </article>
        <article
          data-box-index="9"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #9
          </p>
        </article>
        <article
          data-box-index="10"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #10
          </p>
        </article>
        <article
          data-box-index="11"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #11
          </p>
        </article>
        <article
          data-box-index="12"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #12
          </p>
        </article>
        <article
          data-box-index="13"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; margin-right: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #13
          </p>
        </article>
        <article
          data-box-index="14"
          style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px; background-color: rgb(255, 255, 255); width: 200px; height: 200px;"
        >
          <p
            class="m-u--no-margin"
          >
            Carousel item #14
          </p>
        </article>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;
